manifests/server.pp

```diff
@@ -0,0 +1,78 @@
+# @summary A short summary of the purpose of this class
+#
+# A description of what this class does
+#
+# @param additional_sshd_match_params
+#   Hash of additional sshd match parameters.
+#   Passed to `sshd::allow_from` defined type from `profile_xdmod::server::allow_client`.
+#
+# @param clients
+#   This is a hash that contains all the parameters for `profile_xdmod::server::allow_client`:
+#   ```yaml
+#   profile_xdmod::server::clients:
+#     client1:
+#       hostname: "client1.local"
+#       ip: "172.1.2.3"
+#       ssh_key_pub: "AAAAB..."  # PUBLIC KEY
+#       ssh_key_type: "ssh-rsa"  # ENCRYPTION TYPE
+#   ```
+#
+# @param gid
+#   Group id of user that owns backup files.
+#
+# @param groupname
+#   Groupname that owns backup files.
+#
+# @param slurm_log_parent_dir
+#   Parent directory path where Slurm logs to be shredded are stored
+#
+# @param uid
+#   User id of user that owns backup files.
+#
+# @param username
+#   Username that owns backup files and allowed access.
+#
+# @example
+#   include profile_xdmod::server
+class profile_xdmod::server (
+  Hash   $additional_sshd_match_params,
+  Hash   $clients,
+  String $gid,
+  String $groupname,
+  String $slurm_log_parent_dir,
+  String $uid,
```

Comment:
Can this be an Integer instead of a string?

Reply:
Fixed